bread/layout/components/forms/queryfield.py

```diff
@@ -0,0 +1,201 @@
+import json
+
+import htmlgenerator as hg
+from django.urls import reverse
+from django.utils.translation import gettext_lazy as _
+from djangoql.schema import DjangoQLSchema
+from djangoql.serializers import DjangoQLSchemaSerializer
+
+from bread.layout.components.icon import Icon
+
+
+class BrowseViewSearch(hg.DIV):
+    def __init__(
+        self,
+        model=None,
+        advancedmode=False,
+        size="xl",
+        defaultvalue="",
+        placeholder=None,
+        widgetattributes=None,
+        disabled=False,
+        **kwargs,
+    ):
+        kwargs["_class"] = kwargs.get("_class", "") + f" bx--search bx--search--{size}"
+        kwargs["data_search"] = True
+        kwargs["role"] = "search"
+        width = kwargs.get("width", None)
+        if width:
+            kwargs["style"] = kwargs.get("style", "") + f"width:{width};"
+
+        widgetattributes = {
+            "autocomplete": "off",
+            "autofocus": True,
+            "_class": "bx--search-input",
+            "name": "q",
+            "onfocus": "this.setSelectionRange(this.value.length, this.value.length);",
+            **(widgetattributes or {}),
+        }
+
+        normal_inputid = "normal_search__" + hg.html_id(self)
+        normal_input = hg.INPUT(
+            id=normal_inputid,
+            type="text",
+            value=defaultvalue,
+            placeholder=placeholder or _("Search"),
+            **widgetattributes,
+        )
+        normal_icon = Icon(
+            "search",
+            size=16,
+            _class="bx--search-magnifier",
+            id="searchinputicon_normal",
+            aria_hidden="true",
+        )
+        advanced_inputid = "advanced_search__" + hg.html_id(self)
+        advanced_input = hg.If(
+            model,
+            hg.TEXTAREA(
```

Comment:
This can also be a normal input element, no need for it to be a textarea.

Reply:
Okay, I saw that DjangQL enforces TEXTARE.

Reply:
That's right.